grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -328,6 +329,54 @@ class QueryService implements Status {
         results as Collection<ConsentCollectionLink>
     }
 
+    @SuppressWarnings(["GroovyAssignabilityCheck"])
+    Map<ConsentCollectionLinkDTO, List<StorageDocument>> findCollectionLinksByConsentKeyAndProjectKey(String consentKey, String projectKey) {
+        Map<ConsentCollectionLink, List<StorageDocument>> sampleInfo = new HashMap<>()
+        final session = sessionFactory.currentSession
+        List <ConsentCollectionLink> consentCollectionLinkList = new ArrayList<>()
+        List <BigInteger> consentCollectionIds = new ArrayList<>()
+        final String query =
+                ' select c.id id, c.consent_key consentKey, c.project_key projectKey, c.pii pii, c.compliance compliance, c.sharing_type sharingType , c.text_sharing_type textSharingType,  ' +
+                        ' c.text_compliance textCompliance, c.require_mta requireMta, c.sample_collection_id sampleCollectionId, ' +
+                        ' sc.name collectionName, ic.summary consentName, ip.summary projectName, c.international_cohorts internationalCohorts ' +
+                        ' from consent_collection_link c ' +
+                        ' inner join issue ic on ic.project_key = c.consent_key ' +
+                        ' inner join issue ip on ip.project_key = c.project_key ' +
+                        ' left join sample_collection sc on sc.collection_id = c.sample_collection_id' +
+                        ' where c.project_key = :projectKey' +
+                        ' and c.consent_key = :consentKey'
+        List<ConsentCollectionLinkDTO> results = session.createSQLQuery(query)
+                .setResultTransformer(Transformers.aliasToBean(ConsentCollectionLinkDTO.class))
+                .setString('projectKey', projectKey)
+                .setString('consentKey', consentKey)
+                .list()
+        results.collect {
+            consentCollectionIds.add(it.id)
+            consentCollectionLinkList.add(it)
+        }
+        Map<BigInteger, List <StorageDocument>> storageDocuments = findAllDocumentsBySampleCollectionId(consentCollectionIds)
+        consentCollectionLinkList.each {
+            sampleInfo.put(it, storageDocuments.get(it.id))
+        }
+        sampleInfo
+    }
+
+    @SuppressWarnings(["GroovyAssignabilityCheck"])
+    Map<BigInteger, List <StorageDocument>> findAllDocumentsBySampleCollectionId(List <Integer> consentCollectionIds) {
+        final session = sessionFactory.currentSession
+        final String query =
+                ' select * ' +
+                ' from storage_document ' +
+                ' where consent_collection_link_id in :consentCollectionIds '
+        final SQLQuery sqlQuery = session.createSQLQuery(query)
+        final results = sqlQuery.with {
+            addEntity(StorageDocument)
+            setParameterList('consentCollectionIds', consentCollectionIds)
+            list()
+        }
+        results.groupBy { it.consentCollectionLinkId as BigInteger }
```

Comment:
I'm a little confused by the cast ... the value is a `Long` on `StorageDocument` and should have been a `Long` value when coerced with `addEntity(StorageDocument)`. 

I think maybe there's something funky going on with `.setResultTransformer(Transformers.aliasToBean(ConsentCollectionLinkDTO.class))` because that's where I think you are expecting a `BigInteger`? 

In any case, this is a minor point - I'm OK with it as is. It would be nice if the typing could be cleaned up, but don't let that hold up this PR.

Reply:
IMO: I think we can work on that on https://broadinstitute.atlassian.net/browse/BTRX-686, we have to refactor all this code, we will show only one collection per time, so results will contain only one result